app/uk/gov/hmrc/mtdsatestsupportapi/controllers/requestParsers/validators/CreateTestBusinessValidator.scala

```diff
@@ -90,32 +90,88 @@ class CreateTestBusinessValidator @Inject() (clock: Clock, featureSwitches: Feat
     val errors =
       validateMissingPostcode(business) ++
         validateAccountingPeriod(business) ++
+        validateBusinessAddress(business) ++
+        validateTradingName(business) ++
         business.commencementDate.map(validateCommencementDate).getOrElse(Nil)
 
     errorsResult(errors)
   }
 
-  private def validateMissingPostcode(business: Business): List[MtdError] = {
-    (business.businessAddressPostcode, business.businessAddressCountryCode) match {
-      case (None, Some("GB")) => List(MissingPostcodeError)
-      case _                  => NoValidationErrors
+  private def validateMissingPostcode(business: Business): Seq[MtdError] = {
+    if (featureSwitches.isEnabled("release5")) {
+      if (requiresBusinessPostcode(business) && business.businessAddressPostcode.isEmpty) {
+        Seq(MissingPostcodeError)
+      } else {
+        Nil
+      }
+    } else {
+      (business.businessAddressPostcode, business.businessAddressCountryCode) match {
+        case (None, Some("GB")) => List(MissingPostcodeError)
+        case _                  => NoValidationErrors
+      }
     }
   }
 
-  private def validateAccountingPeriod(business: Business): Seq[MtdError] =
-    if (featureSwitches.isEnabled("release5"))
+  private def validateAccountingPeriod(business: Business): Seq[MtdError] = {
+    if (featureSwitches.isEnabled("release5")) {
       (business.firstAccountingPeriodStartDate, business.firstAccountingPeriodEndDate) match {
         case (Some(start), Some(end)) => TaxYearAlignmentDateRangeValidation.validate(start, end, RuleFirstAccountingDateRangeInvalid)
         case (None, Some(_))          => Seq(MissingFirstAccountintPeriodStartDateError)
         case (Some(_), None)          => Seq(MissingFirstAccountintPeriodEndDateError)
         case _                        => Nil
       }
-    else Nil
+    } else {
+      Nil
```

Comment:
Did you consider using `NoValidationErrors` over Nil (just for readability purposes)?

Reply:
I looked at both and found the NoValidationErrors constant non-obvious (since IIRC it has to be imported) and unnecessary.

If I _only_ returned Nil, I would consider doing something to explain it. But since right above it we have a Seq of obvious errors, I didn't see the point of explaining that an empty Seq of errors means no errors.

Do you still want me to change all of the `Nil` in this file to the constant? From the few validation methods I've seen, Nil seemed to be more common.

Reply:
I mean I suppose `NoValidationErrors` is there for this purpose. But not much in it either way.